src/store/modules/conversations.js

```diff
@@ -297,8 +310,9 @@ export default {
         }
       }
     },
-    setCursor (state, { conversationId, cursor }) {
+    setCursor (state, { conversationId, cursor, first = false }) {
       Vue.set(state.cursors, conversationId, cursor)
+      if (first) Vue.set(state.entries[conversationId], 'firstCursor', cursor)
```

Comment:
`state.entries[conversationId]` gets overwritten pretty often, e.g. when somebody writes a new message in that conversation, the server sends out that update via websocket (for updating `unreadMessageCount`), which triggers `setConversation`. So my guess is that `firstCursor` will be undefined even when it shouldn't be. I didn't confirm this behavior though.

Also, I quite often get this error when loading pickup conversations:

```
Cannot set reactive property on undefined, null, or primitive value: undefined
TypeError: right-hand side of 'in' should be an object, got undefined
Stack trace:
set@webpack-internal:///5:1032:1
setCursor@webpack-internal:///328:718:9
...
```
Seems that `state.entries[conversationId]` is undefined, I just wonder why...

Reply:
Yeah, the `firstCursor` was a bit of a quick hack really. I think I would actually be up for loading this kind of data on entering the route, and clearing it when leaving, but that needs more change.

Another idea would be to store each cursor inside the message too, so when we want to clear out some conversation, can find a point at which we have the cursor available to get the next entries.

Or keep all cursors aroun in the `state.cursors` storage so we can find them again when needed. Probably with a reference to the message id they correspond to.

It seems `state.entries[conversationId]` could be undefined because the messages and conversation are loaded in parallel [1], so if the messages "win" the race, it'll try and set something on the non-existant conversation object.

Although it currently loads the conversation twice, once through the pickup related endpoint, and once through the conversation endpoint, so might be better to only load it in the `conversations` module via a `fetchConversationForPickup()` method.

[1] from `detail.js`
```js
await Promise.all([
  dispatch('conversations/fetchConversation', conversation.id, { root: true }),
  dispatch('conversations/fetch', conversation.id, { root: true }), // gets the messages
])
```